carsus/io/cmfgen/util.py

```diff
@@ -0,0 +1,332 @@
+
+    return phixs_table
+
+
+def get_hydrogenic_nl_phixs_table(hyd_phixs_energy_grid_ryd, hyd_phixs, threshold_energy_ryd, n, l_start, l_end, nu_0=None):
+    """
+    """
+
+    assert l_start >= 0
+    assert l_end <= n - 1
+
+    energy_grid = hyd_phixs_energy_grid_ryd[(n, l_start)]
+    phixs_table = np.empty((len(energy_grid), 2))
+
+    threshold_energy_ev = threshold_energy_ryd * RYD_TO_EV
+    scale_factor = 1 / threshold_energy_ryd / (n ** 2) / ((l_end - l_start + 1) * (l_end + l_start + 1))
+
+    for i, energy_ryd in enumerate(energy_grid):
+        energy_div_threshold = energy_ryd / energy_grid[0]
+        if nu_0 is None:
+            U = energy_div_threshold
+        else:
+            E_0 = (nu_0 * 1e15 * H_IN_EV_SECONDS)
+            U = threshold_energy_ev * energy_div_threshold / (E_0 + threshold_energy_ev)
+        if U > 0:
+            cross_section = 0.0
+            for l in range(l_start, l_end + 1):
+                assert np.array_equal(hyd_phixs_energy_grid_ryd[(n, l)], energy_grid)
+                cross_section += (2 * l + 1) * hyd_phixs[(n, l)][i]
+            cross_section = cross_section * scale_factor
+        else:
+            cross_section = 0.0
+
+        phixs_table[i][0] = energy_div_threshold * threshold_energy_ryd
+        phixs_table[i][1] = cross_section
+
+    return phixs_table
+
+
+def get_opproject_phixs_table(threshold_energy_ryd, a, b, c, d, e, n_points=1000):
+    """
+    References:
+
+        Atomic data for opacity calculations. IX. The lithium isoelectronic 
+        sequence.
+
+        Peach, G. ; Saraph, H. E. ; Seaton, M. J.
+
+        Journal of Physics B: Atomic, Molecular, and Optical Physics, 
+        Volume 21, Issue 22, pp. 3669-3683 (1988).
+    """
+    energy_grid = np.linspace(0.0, 1.0, n_points, endpoint=False)
+    phixs_table = np.empty((len(energy_grid), 2))
+
+    for i, c in enumerate(energy_grid):
+
+        energy_div_threshold = 1 + 20 * (c ** 2)
+        u = energy_div_threshold
+        x = np.log10(min(u, e))
+
+        cross_section = 10 ** (a + x * (b + x * (c + x * d)))
+        if u > e:
+            cross_section *= (e / u) ** 2
+
+        phixs_table[i] = energy_div_threshold * threshold_energy_ryd, cross_section
+
+    return phixs_table
+
+
+def get_hummer_phixs_table(threshold_energy_ryd, a, b, c, d, e, f, g, h, n_points=1000):
+    """
+    References:
+
+        A Fast and Accurate Method for Evaluating the Nonrelativistic Free-free
+        Gaunt Factor for Hydrogenic Ions.
+        
+        Hummer, D. G.
+
+        Astrophysical Journal v.327, p.477
+    """
+
+    # Only applies to `He`. The threshold cross sections seems ok, but 
+    # energy dependence could be slightly wrong. What is the `h` parameter
+    # that is not used?.
+    
+    energy_grid = np.linspace(0.0, 1.0, n_points, endpoint=False)
+    phixs_table = np.empty((len(energy_grid), 2))
+
+    for i, c in enumerate(energy_grid):
+        energy_div_threshold = 1 + 20 * (c ** 2)
```

Comment:
```suggestion
    for i, c1 in enumerate(energy_grid):
        energy_div_threshold = 1 + 20 * (c1 ** 2)
```

Reply:
Otherwise this overwrites the fit coefficient c yielding incorrect results.

Reply:
Done. I used the same name convention from `artisatomic`.